tests/cli/test_cli_metadata.py

```diff
@@ -414,3 +419,125 @@ def test_metadata_deprecate_no_metadata(self, runner):
+                bigquery.SchemaField("column_2", "STRING", description="description 3"),
+            ]
+            result = validate_shredder_mitigation(
+                self.test_path, metadata_from_file, schema
+            )
+            captured = capfd.readouterr()
+            assert result is False
+            assert (
+                "Shredder mitigation validation failed, column_1 does not have a description"
+                " in the schema."
+            ) in captured.out
+
+    def test_validate_shredder_mitigation_group_by(self, runner, capfd):
+        """Test that validation fails and prints a notification when group by contains numbers."""
+        metadata = {
+            "friendly_name": "Test",
+            "labels": {"shredder_mitigation": "true"},
+        }
+
+        schema = [
+            bigquery.SchemaField("column_1", "STRING"),
+            bigquery.SchemaField("column_2", "STRING", description="description 2"),
+        ]
+
+        with runner.isolated_filesystem():
+            query_path = Path(self.test_path) / "query.sql"
+            metadata_path = Path(self.test_path) / "metadata.yaml"
+            os.makedirs(self.test_path, exist_ok=True)
+
+            with open(query_path, "w") as f:
+                f.write("SELECT column_1 FROM test_table GROUP BY ALL")
+            with open(metadata_path, "w") as f:
+                f.write(yaml.safe_dump(metadata))
+
+            metadata_from_file = Metadata.from_file(metadata_path)
+            result = validate_shredder_mitigation(
+                self.test_path, metadata_from_file, schema
+            )
+            captured = capfd.readouterr()
+            assert result is False
+            assert (
+                "Shredder mitigation validation failed, GROUP BY must use an explicit list of"
+                " columns. Avoid expressions like `GROUP BY ALL` or `GROUP BY 1, 2, 3`."
+            ) in captured.out
+
+            with open(query_path, "w") as f:
+                f.write("SELECT column_1 FROM test_table GROUP BY column_1, 2")
+            result = validate_shredder_mitigation(
+                self.test_path, metadata_from_file, schema
+            )
+            captured = capfd.readouterr()
+            assert result is False
+            assert (
+                "Shredder mitigation validation failed, GROUP BY must use an explicit list "
+                "of columns. Avoid expressions like `GROUP BY ALL` or `GROUP BY 1, 2, 3`."
+            ) in captured.out
+
+    def test_validate_metadata_without_labels(self, runner, capfd):
+        """Test that metadata validation doesn't fail when labels are not present."""
+        metadata = {
+            "friendly_name": "Test",
+            "labels": {},
+        }
+        schema = """
+            fields:
+              - mode: NULLABLE
+                name: column_1
+                type: DATE
+                description: description 1
```

Comment:
I think this should be a case that would fail if the label was present so that it verifies the code path isn't run at all.
```suggestion
```

Reply:
Makes sense